job.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const fs = require('fs-extra');
+const _ = require('lodash');
+const DummyLogger = require('sq-logger/dummy-logger');
+const Duration = require('sq-toolkit/duration');
+
+const template = require('./lib/utils/template');
+const BigQueryError = require('./error');
+const BigQueryHelper = require('./helper');
+
+const COST_THRESHOLD_IN_GB = 100;
+
+class BigQueryJob {
```

Comment:
this class behave as singleton, Why don't use the singleton style code that we usually implement (with createAndInitializeInstance, getInstance, clearInstance, etc)

Reply:
This class is supposed to be extended by custom queries job, making overrides of `getQueryParams` and `getQuerySQLDebug`. Also, passing the `sqlFilename` in the constructor in a fixed way